### PR TITLE
Add new socket V2 methods

### PR DESCRIPTION
**Pull request by riccardopillar**

### CODE REVIEW

File: src/sdk/dto/get-cross-chain-bridge-route.dto.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { IsBoolean, IsPositive, IsOptional } from 'class-validator';
+
+export class GetCrossChainBridgeRouteDto {
+  // @IsAddress()
```
Comment (rafaelcidade) [inline src/sdk/dto/get-cross-chain-bridge-route.dto.ts:4]: commented out?

File: src/sdk/exchange/classes/cross-chain-bridge-chain-currency.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { IsNumber } from 'class-validator';
+import { BigNumber } from 'ethers';
+import { BaseClass, TransformBigNumber } from '../../common';
+
+export class CrossChainBridgeChainCurrency extends BaseClass<CrossChainBridgeChainCurrency> {
+  address: string;
```
Comment (rafaelcidade) [inline src/sdk/exchange/classes/cross-chain-bridge-chain-currency.ts:6]: `IsAddress()` ?

File: src/sdk/exchange/classes/cross-chain-bridge-chain-currency.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+  icon: string;
+
+  name: string;
+
+  symbol: string;
+
+  @IsNumber()
```
Comment (rafaelcidade) [inline src/sdk/exchange/classes/cross-chain-bridge-chain-currency.ts:14]: Is class validation needed here?

File: src/sdk/exchange/classes/cross-chain-bridge-route-transaction.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  icon: string;
+  chainAgnosticId?: string;
+  rank: number;
+  updatedAt: string;
+  isEnabled: boolean;
+  createdAt: string;
+}
+
+export class CrossChainBridgeGasFee {
+  @IsNumber()
```
Comment (rafaelcidade) [inline src/sdk/exchange/classes/cross-chain-bridge-route-transaction.ts:26]: Is class validation needed here?

File: src/sdk/exchange/classes/cross-chain-bridge-route-transaction.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  fromAsset: CrossChainBridgeBridgeAsset;
+  protocol: Protocol;
+}
+
+export class CrossChainBridgeTransaction extends BaseClass<CrossChainBridgeTransaction> {
+  userTxType: string;
+
+  txType: string;
+
+  @IsNumber()
```
Comment (rafaelcidade) [inline src/sdk/exchange/classes/cross-chain-bridge-route-transaction.ts:63]: Is class validation needed here?

File: src/sdk/exchange/classes/cross-chain-bridge-route-transaction.ts [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+  icon: string;
+}
+
+export class Step extends BaseClass<Step> {
+  type: string;
+  chainId?: number;
+  fromChainId?: number;
+  fromAmount: string;
+  toAmount: string;
+  protocolFees: CrossChainBridgeProtocolFees;
```
Comment (rafaelcidade) [inline src/sdk/exchange/classes/cross-chain-bridge-route-transaction.ts:51]: Missing `@Type(() => CrossChainBridgeProtocolFees)` ?

(same for the other properties)

File: src/sdk/exchange/classes/cross-chain-bridge-route.ts [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+  chainGasBalances: JSON;
+
+  totalUserTx: number;
+
+  sender: string;
+
+  totalGasFeesInUsd: number;
+
+  userTxs: JSON; //CrossChainBridgeTransaction[]
```
Comment (rafaelcidade) [inline src/sdk/exchange/classes/cross-chain-bridge-route.ts:38]: WIP?

File: src/sdk/exchange/classes/cross-chain-bridge-supported-chain.ts [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+  @IsBoolean()
+  sendingEnabled: boolean;
+
+  icon: string;
+
+  @IsBoolean()
+  receivingEnabled: boolean;
+
+  @Type(() => CrossChainBridgeChainCurrency)
```
Comment (rafaelcidade) [inline src/sdk/exchange/classes/cross-chain-bridge-supported-chain.ts:22]: Is this needed here?
It is no consistent with other classes, for instance:
`cross-chain-bridge-route.ts` and `cross-chain-bridge-route-transaction.ts` have similar properties but decorator

File: src/sdk/exchange/exchange.service.ts [hunk 9]
```diff
@@ -98,4 +109,303 @@ export class ExchangeService extends Service {
...
+        },
+      },
+    );
+
+    return result ? result.items : null;
+  }
+
+  async buildCrossChainBridgeTransaction(dto: CrossChainBridgeRoute): Promise<CrossChainBridgeBuildTXResponse> {
+    const { apiService } = this.services;
+    console.log(dto);
```
Comment (SaltyCucumber) [inline src/sdk/exchange/exchange.service.ts:352]: no consoles, there are two more

**Review by SaltyCucumber — CHANGES_REQUESTED** [review]

File: src/sdk/dto/get-cross-chain-bridge-route.dto.ts [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+  fromTokenAddress: string;
+
+  fromChainId?: number;
```
[the remark this answers is not in the document]
Comment (riccardopillar) [reply src/sdk/dto/get-cross-chain-bridge-route.dto.ts:8]: done

File: src/sdk/exchange/classes/cross-chain-bridge-chain-currency.ts [hunk 11]
```diff
@@ -0,0 +1,18 @@
+import { IsNumber } from 'class-validator';
```
Comment (SaltyCucumber) [inline src/sdk/exchange/classes/cross-chain-bridge-chain-currency.ts:1]: remove

**Review by SaltyCucumber — APPROVED** [review]

**Pull request merged**